### PR TITLE
Modern cmake

### PR DESCRIPTION
**Pull request by bareya**
* Remove Find* cmakes, rely on `Cycles::Cycles` target that is exported in [this](https://github.com/tangent-opensource/cycles/pull/18) PR
* rez variables are optional
* minimise current cmake

### CODE REVIEW

**Comment by bareya** [comment]
https://github.com/tangent-opensource/cycles/pull/18

**Comment by skwerner** [comment]
Need a little time to look at it closer, but I like the reduction in complexity.

**Comment by bareya** [comment]
@skwerner after you review it I would like to release it with cycles latest changes.

File: plugin/hdCycles/renderDelegate.cpp [hunk 1]
```diff
@@ -39,7 +39,11 @@
 #include <render/film.h>
 #include <render/integrator.h>
 
+#if BOOST_NS == hboost
```
Comment (boberfly) [inline plugin/hdCycles/renderDelegate.cpp:42]: We don't need this header at all in renderDelegate.cpp I can't see it being used.

File: plugin/hdCycles/api.h [hunk 2]
```diff
@@ -20,6 +20,8 @@
 #ifndef HD_CYCLES_API_H
 #define HD_CYCLES_API_H
 
+#include <embree3/rtcore.h>
```
Comment (boberfly) [inline plugin/hdCycles/api.h:23]: This isn't too elegant, I am using this as a temporary solution as well, but I think we should think about fixing it in cycles itself.

File: plugin/ndrCycles/api.h [hunk 3]
```diff
@@ -20,6 +20,8 @@
 #ifndef NDR_CYCLES_API_H
 #define NDR_CYCLES_API_H
 
+#include <embree3/rtcore.h>
```
Comment (boberfly) [inline plugin/ndrCycles/api.h:23]: Same here: This isn't too elegant, I am using this as a temporary solution as well, but I think we should think about fixing it in cycles itself.

**Review by boberfly — COMMENTED** [review]
Hey @bareya I like this modern cmake future, less is more!

File: plugin/hdCycles/api.h (hunk 2 above)
Comment (bareya) [reply plugin/hdCycles/api.h:23]: I know it isn't, I am not very happy about it, but that is how I enforce include search path for embree. I agree we should fix it one day on cycles end.

File: plugin/hdCycles/renderDelegate.cpp (hunk 1 above)
Comment (bareya) [reply plugin/hdCycles/renderDelegate.cpp:42]: I didn't look into code itself, just wanted to limit usage of pre-processor defines. I will have a look into this one.

**Comment by bareya** [comment]
@boberfly Thanks!

**Pull request merged**